runtime/doc/lua.txt

```diff
@@ -2513,6 +2513,180 @@ vim.loader.reset({path})                                  *vim.loader.reset()*
       • {path}  (`string?`) path to reset
 
 
+==============================================================================
+Lua module: vim.pack                                                *vim.pack*
+
+WORK IN PROGRESS built-in plugin manager! Early testing of existing features
+is appreciated, but expect breaking changes without notice.
+
+Manages plugins only in a dedicated |packages| *vim.pack-directory*:
+`$XDG_DATA_HOME/nvim/site/pack/core/opt`. Plugin's subdirectory name matches
+plugin's name in specification. It is assumed that all plugins in the
+directory are managed exclusively by `vim.pack`.
+
+Uses Git to manage plugins and requires present `git` executable of at least
+version 2.36. Target plugins should be Git repositories with versions as named
+tags following semver convention `v<major>.<minor>.<patch>`.
+
+Example workflows ~
+
+Basic install and management:
+• Add |vim.pack.add()| call(s) to 'init.lua': >lua
+
+    vim.pack.add({
+      -- Install as "plugin1"/"plugin2" and use greatest available version
+      'https://github.com/user/plugin1',
+      { source = 'https://github.com/user/plugin2' },
+
+      -- Specify plugin's name
+      { source = 'https://github.com/user/generic-name', name = 'plugin3' },
+
+      -- Specify version to follow during install and update
+      {
+        source = 'https://github.com/user/plugin4',
+        version = vim.version.range('1.0'), -- Greatest version matching range
+      },
+      {
+        source = 'https://github.com/user/plugin5',
+        version = 'main', -- Git branch, tag, or commit hash
+      },
+    })
+
+    -- Plugin's code can be used directly after `add()`
+    plugin1 = require('plugin1')
+<
+• Restart Nvim. Plugins that were not yet installed will be available on disk
+  in target state after `add()` call.
+• To update all plugins with new changes:
+  • Execute |vim.pack.update()|. This will download updates from source and
+    show confirmation buffer in a separate tabpage.
+  • Review changes. To confirm all updates execute |:write|. To discard
+    updates execute |:quit|.
```

Comment:
Will this work outside of init.lua, for those of us who still use init.vim? For example, can I use `plugin/pack.lua` or even `lua/pack.lua` (and execute manually with `:lua require('pack')` when I want to modify my plugins)?

One reason I like using `start/` is that I don't have to dedicate any source code to loading plugins. I literally just `git clone` them to the right location and let Nvim handle the rest. Since it sounds like `vim.pack` puts everything in `opt`, I'm assuming that is not a use case that will be supported and I will have to use `vim.pack.add` to load plugins out of `opt`?

Reply:
> that I don't have to dedicate any source code to loading plugins

If you use `vim.pack` wouldn't you then have source code for loading plugins regardless of whether it uses `opt` or `start` internally?

And if you are thinking from a perf pov, using a plugin manager with `opt` will be more efficient than using `start` since you can avoid an unnecessary directory path scan as the package manager knows where to look for plugins.

`start` really only makes sense if you do not want any plugin knowledge in your `init.lua`, and manage them externally, in which case a built-in package manager has no place. Unless I'm missing something?

Reply:
I was a bit careless with my words/phrasing. Yes there will of course always be some code to specify the list of plugins which should be installed (even in the `start` world). I was referring to requiring code to just enable/load the plugin.

But I'm also speaking more from personal preference here. I think in regard to the builtin plugin manager, this design is perfectly fine. And the small minority of users still using init.vim instead of init.lua can use a Lua heredoc or something.

So consider this comment resolved.

Reply:
> Will this work outside of init.lua, for those of us who still use init.vim? For example, can I use `plugin/pack.lua` or even `lua/pack.lua` (and execute manually with `:lua require('pack')` when I want to modify my plugins)?

This should work as any "regular" `vim.xxx()` function, even via `:lua vim.pack.add(...)`.

And yes, 'start/' vs 'opt/' shouldn't make a difference here if they both are required to be listed in the config somewhere.